File: .vs/BookStore/Book_Store/src/app/models/book.ts
export interface Book {
  message: string;
  id: number | 0;               // required, non-nullable
  title: string;            // required
  description: string;      // required
  noOfPages: number;        // required
  isActive: boolean;        // required
  createdOn: string | Date; // backend expects Date, frontend can handle ISO string or Date object
  languageId: number;       // required
  language?: Language;      // optional full Language object
  authorId?: number | null; // optional and nullable - match backend's int? AuthorId
  author?: Author;          // optional full Author object
}
export interface contact {
  id:number;
  message: string;
  name:string;
  email:string;
}

export interface Language {
  id: number;
  name: string;
}

export interface Author {
  id: number;
  name: string;
}

File: .vs/DbOperationWithEfCore/DbOperationsForEfCore/Data/Book.cs
namespace DbOperationsForEfCore.Data
{
    public class Book
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public int NoOfPages { get; set; }
        public bool IsActive { get; set; }
        public DateTime CreatedOn { get; set; }
        public int LanguageId { get; set; }
        public Language? Language { get; set; }
        public int? AuthorId { get; set; }
        public Author? Author { get; set; }
         //Foreign key
        public int? Statusid { get; set; }
        public Status Status { get; set; }



    }
}
